src/shogun/base/SGObject.cpp

```diff
@@ -90,6 +90,17 @@ namespace shogun
 			return map.find(tag) != map.end();
 		}
 
+		ParametersMap filter(ParameterProperties pprop) const {
```

Comment:
this just as an internal filter method?
Sounds like a good idea to me!
Merge?

Reply:
stuff like this should though be hidden from outside (as done here)

Reply:
Yup, should work fine, just need to write a test for it!